src/esp/nav/test/PathFinderTest.cpp

```diff
@@ -120,21 +152,47 @@ void PathFinderTest::pathFindBenchmark() {
     setTestCaseDescription("Path to single point");
     esp::nav::ShortestPath path;
     path.requestedStart = pathFinder.getRandomNavigablePoint();
+    do {
+      path.requestedStart = pathFinder.getRandomNavigablePoint();
+    } while (pathFinder.islandRadius(path.requestedStart) < 10.0);
     path.requestedEnd = pathFinder.getRandomNavigablePoint();
 
     bool status;
     CORRADE_BENCHMARK(5) { status = pathFinder.findPath(path); };
     CORRADE_VERIFY(status);
-  } else {
-    setTestCaseDescription("Path to closest of 100");
+  } else if (testCaseInstanceId() == 1) {
+    setTestCaseDescription("Path to closest of 1000");
     esp::nav::MultiGoalShortestPath path;
-    path.requestedStart = pathFinder.getRandomNavigablePoint();
-    for (int i = 0; i < 100; ++i) {
-      path.requestedEnds.emplace_back(pathFinder.getRandomNavigablePoint());
+    do {
+      path.requestedStart = pathFinder.getRandomNavigablePoint();
+    } while (pathFinder.islandRadius(path.requestedStart) < 10.0);
+
+    std::vector<esp::vec3f> rqEnds;
+    for (int i = 0; i < 1000; ++i) {
+      rqEnds.emplace_back(pathFinder.getRandomNavigablePoint());
     }
+    path.setRequestedEnds(rqEnds);
 
     bool status;
-    CORRADE_BENCHMARK(5) { status = pathFinder.findPath(path); };
+    CORRADE_BENCHMARK(1) { status = pathFinder.findPath(path); };
+    CORRADE_VERIFY(status);
+  } else if (testCaseInstanceId() == 2) {
+    setTestCaseDescription("Cached path to closest of 1000");
+    esp::nav::MultiGoalShortestPath path;
+    do {
+      path.requestedStart = pathFinder.getRandomNavigablePoint();
+    } while (pathFinder.islandRadius(path.requestedStart) < 10.0);
+
+    std::vector<esp::vec3f> rqEnds;
+    for (int i = 0; i < 1000; ++i) {
+      rqEnds.emplace_back(pathFinder.getRandomNavigablePoint());
+    }
+    path.setRequestedEnds(rqEnds);
+
+    pathFinder.findPath(path);
+
+    bool status;
+    CORRADE_BENCHMARK(1) { status = pathFinder.findPath(path); };
     CORRADE_VERIFY(status);
```

Comment:
Since this is three instances and every instance has a dedicated branch, I think it would be cleaner to just have three separate benchmarks :) Alternatively, you could make this data-driven ([canonical example in the docs](https://doc.magnum.graphics/corrade/classCorrade_1_1TestSuite_1_1Tester.html#TestSuite-Tester-instanced)), putting the varying parameters in a struct and fetching them from the test:

```cpp
constexpr struct {
  const char* name;
  bool cache;
  int points;
} PathFindBenchmarkData[]{
  {"path to a single point", false, 1},
  {"path to closest of 1000", false, 1000},
  {"cached path to closest of 1000", true, 1000}
};

...

  addInstancedBenchmarks({&PathFinderTest::pathFindBenchmark},
    100, Cr::Containers::arraySize(PathFindBenchmarkData));
```

and then something like

```cpp
void PathFinderTest::pathFindBenchmark() {
  auto&& data = PathFindBenchmarkData[testCaseInstanceId()];
  setTestCaseDescription(data.name);

  ...

  std::vector<esp::vec3f> rqEnds;
  for (int i = 0; i < data.points; ++i) {
    rqEnds.emplace_back(pathFinder.getRandomNavigablePoint());
  }

  if(data.cached) pathFinder.findPath(path);

  ..
```

Reply:
Did it a little differently as I still want to use the single-goal path object to make sure that code-path is still fast.